extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVault.java

```diff
@@ -39,7 +40,12 @@ public HashicorpVault(@NotNull HashicorpVaultClient hashicorpVaultClient, @NotNu
     public @Nullable String resolveSecret(String key) {
         var result = hashicorpVaultClient.getSecretValue(key);
 
-        return result.succeeded() ? result.getContent() : null;
+        if (result.failed()) {
+            monitor.warning("Failed to resolve secret '%s': %s".formatted(key, result.getFailureMessages()));
```

Comment:
I know that I suggested to have a `warning` log, but thinking about that we would result having the console log bloated with warnings that cannot be resolved because we're often using the pattern of "check into the vault, if it's not present there then look elsewhere". E.g.:
https://github.com/eclipse-edc/Connector/blob/12f1275e791252df14cf02e9fa2c5b884167942e/core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/process/TransferProcessManagerImpl.java#L278-L281

maybe a debug would fit better

Reply:
@ndr-brt I would already stay here with warning. Because as admin you don't have the EDC running in debug mode. And here the message can help, if e.g. the access token is no longer valid or if the rights have changed.

Reply:
@ndr-brt yes, that's a good point. Debug would be my preference.

Reply:
> @ndr-brt I would already stay here with warning. Because as admin you don't have the EDC running in debug mode. And here the message can help, if e.g. the access token is no longer valid or if the rights have changed.

logs should not be used for such activity, the health check service is a better choice, and if I'm correct the hashicorp extension declares some provider on it, so it should be enough.

Reply:
@ndr-brt @jimmarino You both say debug so I would change that to debug, yes?